packages/block-library/src/post-navigation-link/edit.js

```diff
@@ -114,6 +160,22 @@ export default function PostNavigationLinkEdit( {
 					</ToggleGroupControl>
 				</PanelBody>
 			</InspectorControls>
+			<InspectorControls group="advanced">
+				<SelectControl
+					label={ __( 'Filter by taxonomy' ) }
+					value={ taxonomy }
+					options={ getTaxonomyOptions() }
+					onChange={ ( value ) =>
+						setAttributes( {
+							taxonomy: value,
+							inSameTerm: value === '' ? false : true,
+						} )
```

Comment:
The value of `isSameTerm` depends only on whether the value of `taxonomy` is empty. So, can this attribute be removed and not determined during dynamic rendering?